stdlib/importlib/resources/simple.pyi

```diff
@@ -28,11 +27,19 @@ if sys.version_info >= (3, 11):
         def is_file(self) -> Literal[True]: ...
         def is_dir(self) -> Literal[False]: ...
         @overload
-        def open(self, mode: OpenTextMode = "r", *args, **kwargs) -> TextIOWrapper: ...
+        def open(
+            self,
+            mode: Literal["r"] = "r",
+            encoding: str | None = None,
+            errors: str | None = None,
+            newline: str | None = None,
+            line_buffering: bool = False,
+            write_through: bool = False,
+        ) -> TextIOWrapper: ...
         @overload
-        def open(self, mode: OpenBinaryMode, *args: Unused, **kwargs: Unused) -> BinaryIO: ...
+        def open(self, mode: Literal["rb"]) -> BinaryIO: ...
```

Comment:
These overloads are a lot more selective than they used to be. On `main`, the first overload matches if any of these strings is passed in as a literal string:

https://github.com/python/typeshed/blob/0cea0bc06ec33817609080b24691e960fe5774e5/stdlib/_typeshed/__init__.pyi#L164-L199

And similarly, the second overload on `main` matches if any of these strings is passed in as a literal string:

https://github.com/python/typeshed/blob/0cea0bc06ec33817609080b24691e960fe5774e5/stdlib/_typeshed/__init__.pyi#L201-L228

What's the reason for changing it so that the first overload only matches if a literal `"r"` is passed in, and so that the second overload only matches if a literal `"rb"` is passed in?

Reply:
It's now matching [the documentation](https://docs.python.org/3/library/importlib.html#importlib.abc.Traversable.open) of `Traversable`. All other modes make no sense are are likely a bug: https://github.com/python/cpython/blob/dd44ab994b7262f0704d64996e0a1bc37b233407/Lib/importlib/resources/simple.py#L88-L92

(You could argue for allowing `br`, but there's really no case for using this non-standard ordering.)

Reply:
In theory, we could also allow '""' and '"b"', but this is also non-standard according to the docs.

Reply:
Okiedokie